tests/providers/amazon/aws/hooks/test_s3.py

```diff
@@ -313,6 +313,25 @@ def test_copy_object_acl(self, s3_bucket):
             assert ((response['Grants'][0]['Permission'] == 'FULL_CONTROL') and
                     (len(response['Grants']) == 1))
 
+    @mock_s3
+    def test_delete_bucket(self, s3_bucket):
```

Comment:
Can you split this test up into two test cases. One in case the bucket does not exist and one for the regular case - it exists.